File: leetcode/solutions/1189.py
# https://leetcode.com/problems/maximum-number-of-balloons/

class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        res = float("inf")

        freq_dict = {"b": 0, "a": 0, "l": 0, "o": 0, "n": 0}

        for char in text:
            if char in freq_dict:
                freq_dict[char] += 1

        return min(
            freq_dict["b"],
            freq_dict["a"],
            freq_dict["l"] // 2,
            freq_dict["o"] // 2,
            freq_dict["n"]
        )

File: leetcode/solutions/1189.rs
// https://leetcode.com/problems/maximum-number-of-balloons/

use std::collections::HashMap;

impl Solution {
    pub fn max_number_of_balloons(text: String) -> i32 {
        let mut freq_dict: HashMap<char, i32> = HashMap::from([
            ('b', 0),
            ('a', 0),
            ('l', 0),
            ('o', 0),
            ('n', 0)
        ]);

        for chr in text.chars() {
            *freq_dict.entry(chr).or_insert(0) += 1;
        }

        (((freq_dict[&'b']
            .min(freq_dict[&'a']))
            .min(freq_dict[&'l'] / 2))
            .min(freq_dict[&'o'] / 2))
            .min(freq_dict[&'n'])
    }
}
